Sources/NIOPosix/HappyEyeballs.swift

```diff
@@ -70,6 +70,28 @@ public struct NIOConnectionError: Error {
     }
 }
 
+extension NIOConnectionError: CustomStringConvertible {
+    public var description: String {
+        if let dnsError = (dnsAError ?? dnsAAAAError) {
+             return "DNS error: \(dnsError.localizedDescription)"
```

Comment:
Localized description gives bad error messages unless the underlying error is an `NSError`:

```suggestion
             return "DNS error: \(dnsError)"
```

Reply:
Thanks - updated.